indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -146,16 +151,30 @@ def _process_DELETE(self):
     def _process_PATCH(self):
         data = request.json
         updates = {}
-        if set(data.viewkeys()) > {'colors'}:
+        if set(data) - {'colors', 'type_id'}:
             raise BadRequest
         if 'colors' in data:
             colors = ColorTuple(**data['colors'])
             if colors not in get_colors():
                 raise BadRequest
             updates['colors'] = colors
+        if 'type_id' in data:
+            updates.update(self._get_session_type_updates(data['type_id']))
         update_session(self.session, updates)
         return jsonify()
 
+    def _get_session_type_updates(self, type_id):
+        updates = {}
+        if type_id is None:
+            updates['type_id'] = None
+        else:
+            type_ = SessionType.query.with_parent(self.event).filter_by(id=type_id).first()
+            if not type_:
+                raise BadRequest('Invalid type id')
+            if not self.session.type or type_id != self.session.type.id:
+                updates['type_id'] = type_id
```

Comment:
same here; you already query the type so it's better to assign that to the relationship instead of the ID to the FK (which will be done internally by SA on flush)